rules/go_rules.build_defs

```diff
@@ -877,6 +877,22 @@ def go_get(name:str, get:str|list, repo:str='', deps:list=[], exported_deps:list
         else:
             error(f"get_param_as_list: unexpected type %s" % type(value))
 
+    # Given a list of proposed outputs for this rule (which may be paths to .a
+    # files or directories), returns outputs that aren't contained within other
+    # outputs - e.g.:
+    #   remove_redundant_outs(["x/y", "x/z.a", "x/p/q/r", "x/p/q"])
+    # returns
+    #   ["x/y", "x/z.a", "x/p/q"]
+    def remove_redundant_outs(outs):
+        new_outs = {r: True for r in outs}
+        for r in new_outs.keys():
+            dirs = r.split("/")
+            for i in range(1, len(dirs)):
+                root = "/".join(dirs[0:i])
+                if new_outs.get(root):
+                    new_outs[r] = False
+        return [r for r in new_outs.keys() if new_outs[r]]
```

Comment:
To be honest I might move this to the internals of please. If you specify redundant outs, it's really hard to figure out what's going on and this would help out a bit.